modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -439,14 +440,28 @@ INSFVRhieChowInterpolator::finalize()
   std::unordered_map<processor_id_type, std::vector<dof_id_type>> pull_requests;
   static const VectorValue<ADReal> example;
 
-  for (auto * const elem : _elements_to_push_pull)
+  // Create push data
+  for (const auto * const elem : _elements_to_push_pull)
   {
     const auto id = elem->id();
     const auto pid = elem->processor_id();
     auto it = _a.find(id);
     mooseAssert(it != _a.end(), "We definitely should have found something");
     push_data[pid].push_back(std::make_pair(id, it->second));
-    pull_requests[pid].push_back(id);
+  }
+
+  // Create pull data
+  if (_pull_all_nonlocal)
+  {
+    for (const auto * const elem :
+         as_range(_mesh.active_not_local_elements_begin(), _mesh.active_not_local_elements_end()))
```

Comment:
I m curious how we know an element is active if it s not local

same how do we know their processor id or the elem id?

Reply:
If any process has an element then libMesh ensures that the data in that element is complete and uniform across all processes

Reply:
even if the mesh is distributed?

Reply:
correct although in that case many processes may not have a given element at all

Reply:
so the process does not have it, so it doesnt show up in the "non_local" list?

feels like "non_local" is  more "replicated_non_local"

Reply:
To me its implicit in `elements` that the range will not have null/nonexistent elements

Reply:
And `ghosted_non_local` means the same to me as `non_local`

Reply:
But you can debate this name with @roystgnr haha

Reply:
That name is kinda grandfathered in.  Added by John on Sep 7 2007; I started "ParallelMesh" a couple weeks later.

Reply:
> And ghosted_non_local means the same to me as non_local

non_local to me could non_local and (ghosted or non_ghosted). Just not local.